src/ape/api/transactions.py

```diff
@@ -231,10 +233,27 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
         Returns:
             Iterator[:class:`~ape.types.ContractLog`]
         """
-        if not isinstance(abi, EventABI):
-            abi = abi.abi
+        if abi:
+            if not isinstance(abi, EventABI):
+                abi = abi.abi
 
-        yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+            yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+        else:
+            # if abi is not provided, decode all events
+            for log in self.logs:
+                contract_type = self.chain_manager.contracts.get(log["address"])
+                if not contract_type:
+                    logger.warning(f"Failed to locate contract at '{log['address']}'.")
+                    continue
+
+                try:
+                    event_abi = contract_type.events[log["topics"][0]]
+                    yield from self.provider.network.ecosystem.decode_logs(event_abi, [log])
+                except (StopIteration, KeyError, DecodingError):
+                    try:
+                        yield self.provider.network.ecosystem.decode_ds_note(log)  # type: ignore
```

Comment:
Realizing this is an Ethereum specific feature here...

I am actually wondering if this should be in the `EcosystemAPI.decode_logs` method (in ape_Ethereum) and kept as an implementation detail

Reply:
yes, i think it should. it's better if we abstract it into the ecosystem's `decode_logs`